lib/block_view_dao_coin_limit_order.go

```diff
@@ -37,37 +38,135 @@ func (bav *UtxoView) _connectDAOCoinLimitOrder(
+	switch txMeta.DenominatedCoinType {
+	case DAOCoinLimitOrderEntryDenominatedCoinTypeDESO:
+		break
+	case DAOCoinLimitOrderEntryDenominatedCoinTypeDAOCoin:
+		return 0, 0, nil, RuleErrorDAOCoinLimitOrderUnsupportedDenominatedCoinType
+	default:
+		return 0, 0, nil, RuleErrorDAOCoinLimitOrderUnsupportedDenominatedCoinType
+	}
+
+	// If denominated in $DESO, validate DenominatedCoinCreatorPKID is all zeroes.
+	if txMeta.DenominatedCoinType == DAOCoinLimitOrderEntryDenominatedCoinTypeDESO {
+		if !reflect.DeepEqual(ZeroPKID, txMeta.DenominatedCoinCreatorPKID) {
+			return 0, 0, nil, RuleErrorDAOCoinLimitOrderInvalidDAOCoinCreatorPKID
+		}
+	}
+
+	// If denominated in a DAO coin, validate DenominatedCoinCreatorPKID exists and has a profile.
+	if txMeta.DenominatedCoinType == DAOCoinLimitOrderEntryDenominatedCoinTypeDAOCoin {
+		// We currently don't support DAO coins as the denominated type.
+		return 0, 0, nil, RuleErrorDAOCoinLimitOrderUnsupportedDenominatedCoinType
+	}
+
+	// Validate DAOCoinCreatorPKID exists and has a profile.
+	// TODO
+
+	// Validate OperationType is one of our supported enum values.
+	switch txMeta.OperationType {
+	case DAOCoinLimitOrderEntryOrderTypeAsk:
+		break
+	case DAOCoinLimitOrderEntryOrderTypeBid:
+		break
+	default:
+		return 0, 0, nil, RuleErrorDAOCoinLimitOrderUnsupportedOperationType
+	}
+
+	// Validate price > 0.
+	if !txMeta.PriceNanos.Gt(uint256.NewInt()) {
+		return 0, 0, nil, RuleErrorDAOCoinLimitOrderInvalidPrice
+	}
+
+	// If denominated in $DESO, confirm PriceNanos is uint64.
+	if !txMeta.PriceNanos.IsUint64() {
+		return 0, 0, nil, RuleErrorDAOCoinLimitOrderInvalidPrice
+	}
+
+	// If denominated in DAO coins, confirm PriceNanos is uint256.
 	// TODO
 
-	// ----- Begin validations
-	// Create rule errors for each of these validations below
-	// lib/errors.go
-
-	// Validate TransactorPKID exists
-	// Validate DenominatedCoinType is one of our supported enum values and is DESO for now
-	// Validate DenominatedCoinCreatorPKID exists and has a profile or is all zeroes if $DESO
-	// Validate DAOCoinCreatorPKID exists and has a profile
-	// Validate OperationType is one of our supported enum values
-	// Validate PriceNanos > 0
-	//   If denominated in DESO, confirm PriceNanos is uint64
-	//   PriceNanos == uint256.ToUint64(PriceNanos).ToUint256()
-	// Validate Quantity > 0
-	// If DESO buy, validate quantity < max uint64, order.Quantity.IsUint64()
-	// If DESO buy, validate price * quantity < max uint64
+	// Validate quantity > 0.
+	if !txMeta.Quantity.Gt(uint256.NewInt()) {
+		return 0, 0, nil, RuleErrorDAOCoinLimitOrderInvalidQuantity
+	}
+
+	// Validate quantity is uint64.
+	if !txMeta.Quantity.IsUint64() {
+		return 0, 0, nil, RuleErrorDAOCoinLimitOrderInvalidQuantity
+	}
+
+	// If $DESO buy, validate price * quantity < max uint64.
+	if txMeta.DenominatedCoinType == DAOCoinLimitOrderEntryDenominatedCoinTypeDESO {
+		if txMeta.PriceNanos.Uint64()*txMeta.Quantity.Uint64() >= math.MaxUint64 {
```

Comment:
this will just overflow. I think you should do the multiplication in uint256 and then use `IsUint64`